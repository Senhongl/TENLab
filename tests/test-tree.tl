tree = var [123, 0, 123]; # [left, value, right]
tree[0] = var [var [123, 1, 123], 2, var [123, 4, var [123, 5, 123]]];
tree[2] = var [123, 6, 123];

def preorder(t) {
    if (t != 123) {
        print(t[1]);
        preorder(t[0]);
        preorder(t[2]);
    }
    return 1;
}

preorder(tree);