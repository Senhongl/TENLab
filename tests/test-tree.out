0
[ CPUIntType{} ]
2
[ CPUIntType{} ]
1
[ CPUIntType{} ]
4
[ CPUIntType{} ]
5
[ CPUIntType{} ]
6
[ CPUIntType{} ]
